src/abstractdataframe/selection.jl

```diff
@@ -376,12 +376,60 @@ _transformation_helper(df::AbstractDataFrame, col_idx::Int, (fun,)::Ref{Any}) =
 
 _empty_astable_helper(fun, len) = [fun(NamedTuple()) for _ in 1:len]
 
+@noinline _table_transformation(df_sel::AbstractDataFrame, fun) = fun(Tables.columntable(df_sel))
+
+function _sum_fast(cols::Vector{<:AbstractVector})
+    isempty(cols) && throw(ArgumentError("No columns selected for reduction"))
+    return sum(cols)
+end
+
+function _sum_skipmissing_fast(cols::Vector{<:AbstractVector})
+    isempty(cols) && throw(ArgumentError("No columns selected for reduction"))
+    sumz = nothing
```

Comment:
Technically, `zero(eltype(col))` could return `nothing` for weird custom types. I wonder whether `@isdefined` couldn't be used instead.

Reply:
I was thinking about it and concluded it should not happen, as `zero` should return value of type that is passed, but I will fix it for safety.